src/index/coinstatsindex.cpp

```diff
@@ -263,39 +263,21 @@ static bool CopyHeightIndexToHashIndex(CDBIterator& db_it, CDBBatch& batch,
     return true;
 }
 
-bool CoinStatsIndex::CustomRewind(const interfaces::BlockKey& current_tip, const interfaces::BlockKey& new_tip)
+bool CoinStatsIndex::CustomRemove(const interfaces::BlockInfo& block)
 {
     CDBBatch batch(*m_db);
     std::unique_ptr<CDBIterator> db_it(m_db->NewIterator());
 
     // During a reorg, we need to copy all hash digests for blocks that are
     // getting disconnected from the height index to the hash index so we can
     // still find them when the height index entries are overwritten.
-    if (!CopyHeightIndexToHashIndex(*db_it, batch, m_name, new_tip.height, current_tip.height)) {
+    if (!CopyHeightIndexToHashIndex(*db_it, batch, m_name, block.height - 1, block.height)) {
         return false;
     }
 
     if (!m_db->WriteBatch(batch)) return false;
 
-    {
-        LOCK(cs_main);
-        CBlockIndex* iter_tip{m_chainstate->m_blockman.LookupBlockIndex(current_tip.hash)};
-        CBlockIndex* new_tip_index{m_chainstate->m_blockman.LookupBlockIndex(new_tip.hash)};
-        const auto& consensus_params{Params().GetConsensus()};
-
-        do {
-            CBlock block;
-
-            if (!ReadBlockFromDisk(block, iter_tip, consensus_params)) {
-                return error("%s: Failed to read block %s from disk",
-                             __func__, iter_tip->GetBlockHash().ToString());
-            }
-
-            ReverseBlock(block, iter_tip);
-
-            iter_tip = iter_tip->GetAncestor(iter_tip->nHeight - 1);
-        } while (new_tip_index != iter_tip);
-    }
+    ReverseBlock(block);
```

Comment:
d6ebc941f641d68eabe4c3243eb6769dd2d13cf5: this refactor of `CustomRewind ` from iterating over blocks to `CustomRemove` which just takes a single block might be worth doing in a separate commit.

Reply:
re: [#24230 (comment)](https://github.com/bitcoin/bitcoin/pull/24230#discussion_r803655628)

 > [d6ebc94](https://github.com/bitcoin/bitcoin/commit/d6ebc941f641d68eabe4c3243eb6769dd2d13cf5): this refactor of `CustomRewind ` from iterating over blocks to `CustomRemove` which just takes a single block might be worth doing in a separate commit.

I tried this but couldn't actually figure out how how to do it in a good way. I updated the commit description and I want the commit to reflect fact that block read and block undo read is moving from custom indexes up to the BaseIndex::Rewind. If I add the loop in base class without removing it from the custom indexes, it's less clear that code is just moving.